### ISSUE TITLE
Introduce uniform interface for accessing loss function parameters

### ISSUE THREAD

**Issue by ablaom** [JuliaML/LossFunctions.jl#147]
Perhaps I'm missing something, but this does not appear to exist at present. For most losses with a parameter, you access using property, as in 

```julia
loss = HuberLoss(0.5)

julia> loss.d
0.5
```

But for an LPDistLoss there no field for `p`, only a type parameter:

```julia
julia> loss=LPDistLoss(2.5)
LPDistLoss{2.5}()

julia> propertynames(loss)
()
```

One suggestion would be overload `Base.getproperty`/`Base.propertynames` for cases of the second kind (I think these are just `LPDistLoss`, `L1DistLoss`, `L2DistLoss`, but I haven't checked lately), or to implement `StatsBase.params` or both.

I ran into this issue when wrapping the losses for use in MLJ and was obliged to write my own [interface](https://github.com/JuliaAI/MLJBase.jl/blob/a202a689ee3272ce7541ebeeb9a006cec733c4c9/src/measures/loss_functions_interface.jl#L26), which is not robust to extensions of LossFunctions.jl

**Comment by juliohm** [comment]
Yes, instead of fixing things with glue code downstream, it would be much better to have it fixed here. The situation described is not ideal, and we appreciate if you can try to improve the interface here instead of wrapping a new one in MLJ.jl next time 🙏🏽
